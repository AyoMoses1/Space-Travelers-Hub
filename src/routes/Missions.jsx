import React from 'react';
import Table from 'react-bootstrap/Table';
import Badge from 'react-bootstrap/Badge';
import Button from 'react-bootstrap/Button';
import { useDispatch, useSelector } from 'react-redux';
import { changeStatus } from '../redux/missions/missionSlice';

function Missions() {
  const dispatch = useDispatch();
  const { missions } = useSelector((state) => state.missionsReducer);
  return (
    <>
      <Table striped bordered hover responsive>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>g</th>
          </tr>
        </thead>

        {/*  bg={variant.toLowerCase()} */}
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.id}>
              <td>{mission.name}</td>
              <td>{mission.description}</td>
              <td>
                <Badge bg={!mission.reserved ? 'secondary' : 'success'}>
                  {!mission.reserved ? 'NOT A MEMBER' : 'Active Member'}
                </Badge>
              </td>
              <td>
                <div className="center-align">
                  <Button
                    type="button"
                    variant={!mission.reserved ? 'outline-success' : 'outline-danger'}
                    onClick={() => dispatch(changeStatus(mission.id))}
                  >
                    {!mission.reserved ? 'JOIN MISSION' : 'Leave Mission'}
                  </Button>
                </div>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    </>
  );
}

export default Missions;
