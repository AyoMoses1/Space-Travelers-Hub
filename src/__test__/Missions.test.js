import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import Missions from '../routes/Missions';
import missionsReducer from '../redux/missions/missionSlice';

const createTestStore = () => {
  const store = configureStore({
    reducer: {
      missionsReducer,
    },
  });

  return store;
};
export default createTestStore;

describe('Snapshot components test', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    component = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
  });

  test('Missions component test', () => {
    expect(component).toMatchSnapshot();
  });
});