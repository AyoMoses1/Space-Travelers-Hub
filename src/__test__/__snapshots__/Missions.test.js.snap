// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot components test Missions component test 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
